### DISCUSSION TITLE
On defense of tooling

### DISCUSSION THREAD

**Discussion by jonct** [Jip-Hop/jailmaker#237] [Ideas]
> [I'd like to keep it simple. For now no Hatch.](https://github.com/Jip-Hop/jailmaker/pull/233#issuecomment-2231240219) \[[962b238](https://github.com/Jip-Hop/jailmaker/pull/233/commits/962b238f6afde16ce552b9c631ca81e2c9ded7f6)\]

Oh, no no no! 😱 Hatch was *pivotal* to keeping things **simple** for code quality tools in a project with multiple authors.

What Hatch does for us is cleanly wrangle all of the *other* tools for unit tests, regression tests, code coverage reports, static analysis, and style consistency. Immeasurable value; discreet footprint.

None of us wants to deal with all of those separate tools and their interactions on our own, or to have them litter up the repo. VSCode does some of that same work for you privately, but helping others run preflight checks and having ongoing reporting happen **in CI** is invaluable to soliciting other contributions while maintaining a coherent codebase. Please reconsider.

**Comment by Lockszmith-GH** [comment]
That's all good, just deal with it on a new PR, dedicated to making Hatch work. This way other work can continue in parallel.

I made the same mistake with my PRs, they need to be:
1. On topic.
2. As easy as possible to review.

My PR was about changing the script behavior, but I bundled it with my automatic test script.
Splitting them allowed Jip-Hop to deal with each topic separatly.

**Reply by jonct** [reply]
I'm not sure how to continue in parallel, at this point.

I promised not to submit any changes until there's test coverage to prove them. But all unit testing remains blocked on #240.

In case anyone still thinks this is more complicated than it is, I hope you'll find [DEVELOPER.md](https://github.com/Jip-Hop/jailmaker/blob/5c8b41061f6d8c39ed7141b16815645bff3f284d/DEVELOPER.md) reassuring.  🙏

> ## Building Jailmaker
>
> No external dependencies are needed to perform a simple build from the project directory.
>
>     python3 -m scripts.build
> 
> Anything beyond this is *completely optional…*

**Reply by Lockszmith-GH** [reply]
At this point the existing tests complete, so we're just waiting for JipHop to come back from vacation - no?
